src/main/java/com/google/devtools/build/lib/actions/cache/DigestUtils.java

```diff
@@ -331,4 +302,35 @@ public static void write(byte[] digest, OutputStream sink) throws IOException {
     }
     return xor(rhs, lhs);
   }
+
+  /** Utility class for combining a set of (path, metadata) pairs into a single digest. */
+  public static final class OrderIndependentHasher {
```

Comment:
why not add this as a top-level class?

Reply:
I just put it here because it replaces functionality that was in `DigestUtils`. (Also, it shares the private `xor()` method with the outer class, but we could just increase the visibility of that.)

I don't have a strong opinion about its location.

Reply:
I just promoted it to it's own file.